gap/congruences/conglatt.gi

```diff
@@ -7,111 +7,81 @@
-## actual list of lists of integers which defines the containments in the
-## lattice.  Component 2 in the list is the list of congruences, which is hidden
-## when this object is used.
-
-InstallMethod(CongruencesOfLattice,
-"for a congruence lattice",
-[IsCongruenceLattice],
-function(latt)
-  return latt![2];
-end);
 
 InstallMethod(\[\],
-"for a congruence lattice",
-[IsCongruenceLattice, IsObject],
-function(latt, x)
-  return latt![1][x];
+"for a congruence poset",
+[IsCongruencePoset, IsObject],
+function(poset, x)
+  return OutNeighboursOfVertex(poset!.po, x);
 end);
 
 InstallMethod(ViewObj,
-"for a congruence lattice",
-[IsCongruenceLattice],
-function(latt)
-  ViewObj(latt![1]);
+"for a congruence poset",
+[IsCongruencePoset],
+function(poset)
+  ViewObj(OutNeighbours(poset!.po));
 end);
 
 InstallMethod(PrintObj,
-"for a congruence lattice",
-[IsCongruenceLattice],
-function(latt)
-  PrintObj(latt![1]);
+"for a congruence poset",
+[IsCongruencePoset],
+function(poset)
+  PrintObj(OutNeighbours(poset!.po));
 end);
 
 InstallMethod(Length,
-"for a congruence lattice",
-[IsCongruenceLattice],
-function(latt)
-  return Length(latt![1]);
+"for a congruence poset",
+[IsCongruencePoset],
+function(poset)
+  return DigraphNrVertices(poset!.po);
 end);
 
 InstallMethod(IsBound\[\],
-"for a congruence lattice and a positive integer",
-[IsCongruenceLattice, IsPosInt],
-function(latt, x)
-  return IsBound(latt![1][x]);
+"for a congruence poset and a positive integer",
+[IsCongruencePoset, IsPosInt],
+function(poset, x)
+  return x >= 1 and x <= Length(poset);
 end);
 
-###############################################################################
-# LatticeOfXCongruences function
-###############################################################################
-#
-# This abstract function takes a semigroup 'S', a string 'type_string' and a
-# record 'record'.
-# type_string should be in ["Left", "Right", ""] and describes the sort of
-# relations we want to find (respectively: left congruences, right congruences,
-# two-sided congruences), referred to as 'x congs' below.
-# record may contain any of the following components, which should be set to
-# 'true' to have the stated effect:
-#   * minimal - Return only minimal x-congs
-#   * 1gen - Return only x-congs with a single generating pair
-#   * restriction - Return only x-congs generated by relating these elements
-#
-###############################################################################
-
-# FIXME Monolithic functions = BAD!!! JDM
-
-SEMIGROUPS.LatticeOfXCongruences := function(S, type_string, record)
-  local report, _XSemigroupCongruence, elms, pairs, total, congs1, nrcongs,
-        children, parents, last_collected, nr, pair, badcong, newchildren,
-        newparents, newcong, i, c, p, congs, set_func, lattice, join_func,
-        length, found, ignore, start, j, k;
-
+SEMIGROUPS.PrincipalXCongruencePoset := function(S, restriction, cong_func)
```

Comment:
Or maybe better check the arguments of the function/method in the function/method so that it is clear what is required. This might not be feasible or desirable...